### ISSUE TITLE
quarto book .content-visible ignored

### ISSUE THREAD

**Issue by verbalins** [quarto-dev/quarto-cli#2140]
### Bug description

Change the default contents of index.qmd created in a new book project to:

````md
::: {.content-visible when-format="html"}
# Preface {.unnumbered}

This is a Quarto book.

To learn more about Quarto books visit <https://quarto.org/docs/books>.
:::
````
The compiled pdf file shows the chapter heading "Preface" but not the contents (This is a Quarto book...). It is my understanding that the chapter title should also be ignored?

quarto check:
````
[✓] Checking Quarto installation......OK
      Version: 1.1.132
      Path: /opt/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.6
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.2.1
      Path: /usr/lib64/R
      LibPaths:
        - /home/verbalins/R/x86_64-pc-linux-gnu-library/4.2
        - /usr/lib/R/library
      rmarkdown: 2.14

[✓] Checking Knitr engine render......OK
````
quarto render -t all output:
````
1/4] index.qmd
[2/4] intro.qmd
[3/4] summary.qmd
[4/4] references.qmd

pandoc 
  to: latex
  output-file: index.tex
  standalone: true
  toc: true
  number-sections: true
  top-level-division: chapter
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  
metadata
  crossref:
    chapters: true
  documentclass: scrreprt
  papersize: letter
  classoption:
    - DIV=11
    - numbers=noendperiod
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  bibliography:
    - references.bib
  title: repro
  author: Jane Doe
  date: 8/26/2022
  
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999994 (TeX Live 2022/Arch Linux) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
running xelatex - 2
  This is XeTeX, Version 3.141592653-2.6-0.999994 (TeX Live 2022/Arch Linux) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  

Output created: _book/index.html
````

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://yihui.org/issue/#please-format-your-issue-correctly) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
Thanks for the report! I can reproduce the issue. Unfortunately, the issue is not entirely simple to solve because it involves the order in which filters are applied, and we find the sections before applying the `content-hidden` transformation.

**Comment by jjallaire** [comment]
Yes, we do special handing of the initial header within book source files so using content-visible in this fashion won't work.

**Comment by verbalins** [comment]
Thank you for your response! Any suggestions on how to handle this currently? Write a lua filter to skip that part in the pdf?

**Comment by kjohnsen** [comment]
I'm running into this too...would appreciate the help. I have some chapters I need for HTML but need to hide for LaTeX

**Comment by verbalins** [comment]
I actually wrote a lua script to handle this. Let me get back to you about how to do it.

**Comment by verbalins** [comment]
> I'm running into this too...would appreciate the help. I have some chapters I need for HTML but need to hide for LaTeX

https://github.com/verbalins/his-template/blob/main/_extensions/his/ignore_index.lua Heres my lua to remove the index from the pdf.

**Comment by kjohnsen** [comment]
Awesome. Thank you!!!

**Comment by e-miz** [comment]
Stumbled on this and was a bit confused on getting this to work since I don't really have experience writing filters or lua. Realized the example is a bit specific, but it was enough for me to figure out what was going on (thanks for providing it!)

I was able to write something a bit more general that uses quarto's existing syntax to remove headers. (note perfect logic is not guaranteed, but it worked for my brief tests).
 
In case anyone finds it useful:

```lua
function Header(element)
    if (element.classes:includes("content-hidden") and quarto.doc.isFormat(element.attributes["when-format"]))
        or (element.classes:includes("content-visible") and not quarto.doc.isFormat(element.attributes["when-format"])) then
        return {}
    end
end
```

As an example:

This additional filter will delete `Preface`. Everything else inside the div will be removed as expected. This is consistent with the behavior that was mentioned above, which is that the first header is always treated specially in a book project. So in theory you only need to add this to the first header in your document.

```qmd
::: {.content-visible when-format="html"}

# Preface {.unnumbered .content-visible when-format="html"}

Testing Text

# Acknowledgements {.unnumbered}

Coming soon!
:::
```

**Comment by strimmerlab** [comment]
I have a similar problem. I would like to selectively include or exclude whole chapters (both pdf and html) from a book.

````
::: {.content-hidden unless-meta="advancedcontent"}

# Advanced chapter

## Topic 1

:::

````
The main header of the advanced chapter is always included even if `advancedcontent: false`.

How can this be done without removing the advanced chapter  manually from `_quarto.yml`?

**Comment by mcanouil** [comment]
@strimmerlab in your case you should consider using Profile, see <https://quarto.org/docs/projects/profiles.html>.

**Comment by strimmerlab** [comment]
Thanks for the reply.  Yes, that's exactly what I am doing. I am following strictly the example shown in the page you linked to (https://quarto.org/docs/projects/profiles.html) , with a variable `advancedcontent: false` in the profile and a hidden/visible environment.

In both pdf and html the first header is shown regardless whether it is hidden.

**Comment by cscheid** [comment]
> In both pdf and html the first header is shown regardless whether it is hidden.

@mcanouil's suggestion is to use profiles to change the project metadata so that different profiles expose different chapter lists .

**Comment by strimmerlab** [comment]
Many thanks for the clarification!

**Comment by s2t2** [comment]
I can confirm [conditional formatting](https://quarto.org/docs/authoring/conditional.html?utm_source=chatgpt.com) doesn't work as advertised.

I am using this markdown to conditionally display a .gif file on web, or a .png file in the pdf (because a .gif file can't be rendered in a .pdf):

```md
::: {.content-visible when-format="html"}
![List of packages pre-installed in Google Colab](../../../images/pip-list.gif)
:::

::: {.content-visible when-format="pdf"}
![List of packages pre-installed in Google Colab](../../../images/pip-list.png)
:::
```

When previewing the site with `quarto preview`, it only shows the "html" format section, and not the "pdf" section. This is desired.

However when rendering to pdf with `quarto render --to pdf`, quarto erroneously still tries to process the "html" format section. This breaks the ability to render to pdf. And is a big blocker.

```
Rendering PDF
running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999997 (TeX Live 2025/Homebrew) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages
ERROR: 
compilation failed- error
LaTeX Error: Cannot determine size of graphic in notes/python-lang/../../imag
es/vertical-column-select.gif (no BoundingBox).

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.1063 ...../../images/vertical-column-select.gif}
                                                  
Try typing  <return>  to proceed.
If that doesn't work, type  X <return>  to quit.
```

When functionality like this doesn't work as advertised in the docs, this lowers the credibility of the documentation and of the quarto tool in general.
